code/modules/atmospherics/machinery/components/unary_devices/vent_pump.dm

```diff
@@ -21,40 +16,34 @@
 	pipe_state = "uvent"
 	vent_movement = VENTCRAWL_ALLOWED | VENTCRAWL_CAN_SEE | VENTCRAWL_ENTRANCE_ALLOWED
 
-	///Direction of pumping the gas (RELEASING or SIPHONING)
-	var/pump_direction = RELEASING
-	///Should we check internal pressure, external pressure, both or none? (EXT_BOUND, INT_BOUND, NO_BOUND)
-	var/pressure_checks = EXT_BOUND
+	///Direction of pumping the gas (ATMOS_DIRECTION_RELEASING or ATMOS_DIRECTION_SIPHONING)
+	var/pump_direction = ATMOS_DIRECTION_RELEASING
+	///Should we check internal pressure, external pressure, both or none? (ATMOS_EXTERNAL_BOUND, ATMOS_INTERNAL_BOUND, NO_BOUND)
+	var/pressure_checks = ATMOS_EXTERNAL_BOUND
 	///The external pressure threshold (default 101 kPa)
 	var/external_pressure_bound = ONE_ATMOSPHERE
 	///The internal pressure threshold (default 0 kPa)
 	var/internal_pressure_bound = 0
-	// EXT_BOUND: Do not pass external_pressure_bound
-	// INT_BOUND: Do not pass internal_pressure_bound
+	// ATMOS_EXTERNAL_BOUND: Do not pass external_pressure_bound
+	// ATMOS_INTERNAL_BOUND: Do not pass internal_pressure_bound
 	// NO_BOUND: Do not pass either
 
-	///Frequency id for connecting to the NTNet
-	var/frequency = FREQ_ATMOS_CONTROL
-	///Reference to the radio datum
-	var/datum/radio_frequency/radio_connection
-	///Radio connection to the air alarm
-	var/radio_filter_out
-	///Radio connection from the air alarm
-	var/radio_filter_in
-
 /obj/machinery/atmospherics/components/unary/vent_pump/New()
 	if(!id_tag)
 		id_tag = SSnetworks.assign_random_name()
 	. = ..()
 
+/obj/machinery/atmospherics/components/unary/vent_pump/Initialize(mapload)
+	. = ..()
+
+	var/area/area = get_area(src)
+	LAZYADD(area.air_vents, src)
+
 /obj/machinery/atmospherics/components/unary/vent_pump/Destroy()
 	var/area/vent_area = get_area(src)
 	if(vent_area)
-		vent_area.air_vent_info -= id_tag
-		GLOB.air_vent_names -= id_tag
+		vent_area.air_vents -= src
```

Comment:
you need to handle area changing cases for these/all the other things you're linking to area, or have them track the area they're registered in. Otherwise we just get weird harddels